src/services/redis/backend.rs

```diff
@@ -0,0 +1,457 @@
+
+        Ok(size)
+    }
+
+    async fn stat(&self, path: &str, _args: OpStat) -> Result<ObjectMetadata> {
+        let path = path.to_string();
+        let mut con = self
+            .client
+            .get_tokio_connection_manager()
+            .await
+            .map_err(|e| new_get_async_con_error(e, Operation::Stat, path.as_str()))?;
+
+        let m_path = build_abs_meta_path(self.root.as_str(), path.as_str());
+        let bin: Vec<u8> = con
+            .get(m_path)
+            .await
+            .map_err(|e| new_exec_async_cmd_error(e, Operation::Stat, path.as_str()))?;
+
+        bincode::deserialize(&bin[..])
+            .map_err(|e| new_deserialize_metadata_error(e, Operation::Stat, path.as_str()))
+    }
+
+    async fn delete(&self, path: &str, _args: OpDelete) -> Result<()> {
+        let path = path.to_string();
+        let abs_path = build_abs_path(self.root.as_str(), path.as_str());
+
+        let mut con = self
+            .client
+            .get_tokio_connection_manager()
+            .await
+            .map_err(|err| new_get_async_con_error(err, Operation::Delete, path.as_str()))?;
+
+        // a reversed-level-order traversal delete
+        // recursion is not friendly for rust, and on-heap stack is preferred.
+        let (mut queue, mut rev_stack) = (VecDeque::new(), VecDeque::new());
+
+        queue.push_back(path.to_string());
+        while let Some(pop) = queue.pop_front() {
+            let skey = format!("v{}:k:{}", REDIS_API_VERSION, pop);
+            let mut it = con
+                .sscan(skey)
+                .await
+                .map_err(|err| new_exec_async_cmd_error(err, Operation::Delete, path.as_str()))?;
+            while let Some(child) = it.next_item().await {
+                let _ = queue.push_back(child);
+            }
+            let _ = rev_stack.push_back(pop);
+        }
+
+        while let Some(pop) = rev_stack.pop_back() {
+            remove(&mut con, abs_path.as_str(), pop.as_str()).await?;
+        }
+
+        if let Some(parent) = PathBuf::from(abs_path.clone()).parent() {
+            let p = parent.display().to_string();
+            let skey = format!("v{}:k:{}", REDIS_API_VERSION, p);
+            con.srem(skey, abs_path.clone()).await.map_err(|err| {
+                new_exec_async_cmd_error(err, Operation::Delete, abs_path.as_str())
+            })?;
+        }
+
+        Ok(())
+    }
+
+    async fn list(&self, path: &str, _args: OpList) -> Result<DirStreamer> {
+        let path = build_abs_path(self.root.as_str(), path);
+        let con = self
+            .client
+            .get_tokio_connection_manager()
+            .await
+            .map_err(|err| new_exec_async_cmd_error(err, Operation::List, path.as_str()))?;
+        Ok(Box::new(DirStream::new(
+            Arc::new(self.clone()),
+            con,
+            path.as_str(),
+        )))
+    }
+}
+
+fn build_abs_content_path<'a>(root: &'a str, rel_path: &'a str) -> String {
```

Comment:
Please don't embed our functions. Instead, try to compose them.

For example, we can delcare a `V0_CONTENT_PREFIX = "v0:c:"`

And add a function called: `v0_content_prefix(s: &str) -> String`.

So we can use `v0_content_prefix(&build_abs_path(root, rel_path))`.